collect_app/src/test/java/org/odk/collect/android/projects/ProjectDeleterTest.kt

```diff
@@ -105,14 +116,57 @@ class ProjectDeleterTest {
             formUpdateManager,
             instanceSubmitScheduler,
             instancesRepository,
-            ""
+            "",
+            changeLockProvider
         )
 
         val result = deleter.deleteCurrentProject()
         assertThat(result, instanceOf(DeleteProjectResult.DeletedSuccessfully::class.java))
         assertThat((result as DeleteProjectResult.DeletedSuccessfully).newCurrentProject, `is`(nullValue()))
     }
 
+    @Test
+    fun `If there are running background jobs that use blank forms the project should not be deleted`() {
+        val formChangeLock = BooleanChangeLock()
+        formChangeLock.lock()
+
+        whenever(changeLockProvider.getFormLock(any())).thenReturn(formChangeLock)
+
+        val deleter = ProjectDeleter(
+            mock(),
+            currentProjectProvider,
+            formUpdateManager,
+            instanceSubmitScheduler,
+            instancesRepository,
+            "",
+            changeLockProvider
+        )
+
+        val result = deleter.deleteCurrentProject()
+        assertThat(result, instanceOf(DeleteProjectResult.RunningBackgroundJobs::class.java))
+    }
+
+    @Test
+    fun `If there are running background jobs that use saved forms the project should not be deleted`() {
```

Comment:
I don't think we actually need to protect ourselves from this, but at the same time it does nicer code-wise to reference both change locks.